From 1259e7e047d5556f5552d4cf0d07c86165629be1 Mon Sep 17 00:00:00 2001
From: Corinna Vinschen <vinschen@redhat.com>
Date: Mon, 14 Feb 2022 21:34:46 +0100
Subject: [PATCH 1/2] Advertise "uv" capability, not "clientPin".

Per the CTAP protocol the "uv" capability indicates that the device is
capable of verifying the user within itself, while "clientPin" indicates
that the device is capable of accepting a PIN from the client.  WinHello
only supports "uv", but not "clientPin", but current libfido2 advertises
"clientPin", not "uv".  Change this in fido_winhello_get_cbor_info.

Signed-off-by: Corinna Vinschen <vinschen@redhat.com>
---
 src/winhello.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/winhello.c b/src/winhello.c
index 9de6c6c..2fc6a2a 100644
--- a/src/winhello.c
+++ b/src/winhello.c
@@ -883,7 +883,7 @@ fido_winhello_get_cbor_info(fido_dev_t *dev, fido_cbor_info_t *ci)
 	const char *v[3] = { "U2F_V2", "FIDO_2_0", "FIDO_2_1_PRE" };
 	const char *e[2] = { "credProtect", "hmac-secret" };
 	const char *t[2] = { "nfc", "usb" };
-	const char *o[4] = { "rk", "up", "plat", "clientPin" };
+	const char *o[4] = { "rk", "up", "plat", "uv" };
 
 	(void)dev;
 
-- 
2.35.1.windows.2

